app/api/assets_routes.py

```diff
@@ -0,0 +1,365 @@
+import os
+import uuid
+import urllib.parse
+from typing import Optional
+
+from aiohttp import web
+from pydantic import ValidationError
+
+import folder_paths
+
+from .. import assets_manager, assets_scanner
+from . import schemas_in, schemas_out
+
+
+ROUTES = web.RouteTableDef()
+
+
+@ROUTES.head("/api/assets/hash/{hash}")
+async def head_asset_by_hash(request: web.Request) -> web.Response:
+    hash_str = request.match_info.get("hash", "").strip().lower()
+    if not hash_str or ":" not in hash_str:
+        return _error_response(400, "INVALID_HASH", "hash must be like 'blake3:<hex>'")
+    algo, digest = hash_str.split(":", 1)
+    if algo != "blake3" or not digest or any(c for c in digest if c not in "0123456789abcdef"):
+        return _error_response(400, "INVALID_HASH", "hash must be like 'blake3:<hex>'")
+    exists = await assets_manager.asset_exists(asset_hash=hash_str)
+    return web.Response(status=200 if exists else 404)
+
+
+@ROUTES.get("/api/assets")
+async def list_assets(request: web.Request) -> web.Response:
+    query_dict = dict(request.rel_url.query)
+
+    try:
+        q = schemas_in.ListAssetsQuery.model_validate(query_dict)
+    except ValidationError as ve:
+        return _validation_error_response("INVALID_QUERY", ve)
+
+    payload = await assets_manager.list_assets(
+        include_tags=q.include_tags,
+        exclude_tags=q.exclude_tags,
+        name_contains=q.name_contains,
+        metadata_filter=q.metadata_filter,
+        limit=q.limit,
+        offset=q.offset,
+        sort=q.sort,
+        order=q.order,
+    )
+    return web.json_response(payload.model_dump(mode="json"))
+
+
+@ROUTES.get("/api/assets/{id}/content")
```

Comment:
Are we missing `GET /assets/{id}`? Don't see it in the diff.

Without it I would need to use `GET /api/assets` and filter for asset ID i need.

Reply:
I added this in the last commit, thanks for pointing on this